### PR TITLE
Chapter 3/Homeowork 3

### PR DESCRIPTION
**Pull request by Tailor-AI**
First draft of homework 3.

### CODE REVIEW

File: Src/BootCamp.Chapter/Equipment.cs [hunk 1]
```diff
@@ -85,7 +59,44 @@ public float GetTotalDefense()
...
+
+        public override string ToString()
+        {
+            return string.Format($"{EquipmentToString()} and weapon: {Weapon.Name}");
+        }
+
+        private string EquipmentToString()
+        {
+            string result = "";
+            Armour[] armourArray = new Armour[8];
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Equipment.cs]: Armour[] armour = {Head, Chest, LeftSHoulder, RightShoulder, LeftArm, RightArm, Gloves, Legs}

File: Src/BootCamp.Chapter/Items/Armour.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter.Items
+{
+    public class Armour : Item
+    {
+        public float Defense { get; private set; }
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Items/Armour.cs]: Can be readonly. No need setter altogether.

File: Src/BootCamp.Chapter/Player.cs [hunk 3]
```diff
@@ -29,42 +33,104 @@ public class Player
...
         }
 
         public void Remove(Item item)
         {
+            _inventory.RemoveItem(item ?? throw new ArgumentNullException());
+        }
 
+        public float GetTotalInventoryWeight()
+        {
+            return _inventory.Items.Length == 0 ? 0 : _inventory.Items.Sum(x => x.Weight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Player.cs]: You can format slightly more readable:
```cs
return _inventory.Items.Length == 0 
  ? 0 
  : _inventory.Items.Sum(x => x.Weight); 
```

File: Src/BootCamp.Chapter/Shop.cs [hunk 4]
```diff
@@ -1,28 +1,15 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Linq;
+
+namespace BootCamp.Chapter
 {
-    public class Shop
+    public class Shop : Inventory
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Shop.cs]: Shop is not an inventory. That's illogical. Don't inherit just because it's convenient for now.

**Review by Almantask — CHANGES_REQUESTED** [review]

File: Src/BootCamp.Chapter/Equipment.cs (hunk 1 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Equipment.cs]: Fixed. Much cleaner I see. thx

File: Src/BootCamp.Chapter/Items/Armour.cs (hunk 2 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Items/Armour.cs]: Fixed.

File: Src/BootCamp.Chapter/Player.cs (hunk 3 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Player.cs]: Fixed.

File: Src/BootCamp.Chapter/Shop.cs (hunk 4 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Shop.cs]: Ok. I now added new Inventory in the shop instead. However your ShopTest requires that shop has Items (_shop.Items.Should etc.).
The Inventory I added to shop already has Items as a prop but how can I make a reference to this in the Shop directly?
I solved this by adding Items also in the shop class and in the constructor put Items = inventory.Items. Ok?
Of course I had to add this line also everytime I add/remove something in inventory.Items

Comment (Almantask) [reply Src/BootCamp.Chapter/Shop.cs]: exactly. Shop **is not** inventory. Sho **has** inventory!

Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Shop.cs]: Thanks for review!

**Review by Almantask — APPROVED** [review]
Looks good now

**Pull request closed without merging**